OptimoveSDK/Sources/Classes/Optimobile/Optimobile+DeepLinking.swift

```diff
@@ -32,6 +32,20 @@ public enum DeepLinkResolution {
     case linkExpired(_ url:URL)
     case linkLimitExceeded(_ url:URL)
     case linkMatched(_ data:DeepLink)
+    
+    func toDict() -> NSDictionary {
+        let dict = NSDictionary()
+        
+        switch self {
+        case let .lookupFailed(url): dict.setValue(url, forKey: "lookupFailed")
+        case let .linkNotFound(url): dict.setValue(url, forKey: "linkNotFound")
+        case let .linkExpired(url): dict.setValue(url, forKey: "linkExpired")
+        case let .linkLimitExceeded(url): dict.setValue(url, forKey: "linkLimitExceeded")
+        case let .linkMatched(data): dict.setValue(data, forKey: "linkMatched")
```

Comment:
Do we need to go deeper into converting 'data' to a dict as well, I noticed it was a DeepLink struct

Reply:
i have looked at the deeper class׳s and structs and it looks like they confirm to the objc needs

Reply:
OK, you mean we can expose this struct to Objective-C?

I possibly misunderstood but I thought Objective-C can't use struct types from Swift at all--am I mistaken?